### PR TITLE
refactor(cypress): reuse config update command

### PR DESCRIPTION
**Pull request by Ankesh2004**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

This pull request resolves issue #5913 

## Description
This PR refactors the existing Cypress commands used for configuration updates (autoRetryConfig, setMaxAutoRetries, and stepUp) into a single unified command cy.updateConfig. The refactor helps reduce redundancy in the codebase by consolidating multiple similar implementations into one versatile command that can handle configuration changes for auto retries, max retries, and step-up flows.

## Key Changes
-Combined autoRetryConfig, setMaxAutoRetries, and stepUp into the cy.updateConfig command.
-Updated the Retries.cy.js test cases to use the new unified command for disabling/enabling auto retries and step-up configurations.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
The refactor was necessary to reduce code duplication and improve maintainability. By creating a unified command, we ensure that future changes to configuration updates are centralized in one place, making the code easier to manage and extend.

## How did you test it?
Tested manually and with cypress command.

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [x] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/support/commands.js
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:2319]: Can you also define `key` within the case statements?

For `stepUp`: `const key = `step_up_enabled_${merchantId}`;`
For `maxRetries`: `const key = `max_auto_retries_enabled_${merchantId}`;`
For `autoRetries`: `const key = `should_call_gsm_${merchantId}`;`

And yes, this key should be passed in to the `url`. Something like: `${baseUrl}/configs/${key};`

Example:

```js
Cypress.Commands.add("updateConfig", (configType, configData, globalState, value) => {
  const base_url = globalState.get("baseUrl");
  const merchant_id = globalState.get("merchantId");

  let body;  
  let key;
  let url;

    case 'autoRetry':
	  key = `should_call_gsm_${merchant_id}`;
      url = `${baseUrl}/configs/${key}`;
```

File: cypress-tests/cypress/support/commands.js
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:2320]: I have a hunch that this would not work as we only pass `key` and `value` in body for all 3 commands: https://github.com/juspay/hyperswitch/blob/main/cypress-tests/cypress/fixtures/configs.json

So you would have to revert back to passing data something like below:
```js
  body = { configData.key = key, configData.value = value };
```

File: cypress-tests/cypress/support/commands.js
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:2329]: same applies to these 2 as well.

File: cypress-tests/cypress/support/commands.js
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:2331]: Can we have clear error message? This needs to be done just so that, if tomorrow, when Cypress throws this error, we should be able to see what config type was passed:
```suggestion
      throw new Error(`Invalid config type passed into the configs: "${key}: ${value}"`);
```

File: cypress-tests/cypress/support/commands.js
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:2339]: Can you define a `const api_key` above with value `globalState.get("adminApiKey")` and pass the value in here instead?

```suggestion
      "api-key": api_key,
```

File: cypress-tests/cypress/support/commands.js
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:2383]: Please revert this change. We need to log `x-request-id`. This is done because if anything goes wrong and we get an error, this `x-request-id` helps us in tracking down the issue by giving access to the logs.

File: cypress-tests/cypress/support/commands.js
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:2388]: In the `response`, we get what we send in the `request` when the response status is `200`. So it is better have the same..

**Review by pixincreate — COMMENTED** [review]
This is a pretty good approach though! Found a small room for improvement, other than that, changes look almost perfect!

Thanks for the PR!

**Comment by gorakhnathy7** [comment]
Hey @Ankesh2004 kindly look into the comments and resolve them.

**Comment by Ankesh2004** [comment]
> Hey @Ankesh2004 kindly look into the comments and resolve them.

Sure, I will get it done by today 👍

File: cypress-tests/cypress/e2e/RoutingTest/00003-Retries.cy.js
Comment (Ankesh2004) [reply cypress-tests/cypress/e2e/RoutingTest/00003-Retries.cy.js]: Done

File: cypress-tests/cypress/support/commands.js
Comment (Ankesh2004) [reply cypress-tests/cypress/support/commands.js]: Done

**Review by pixincreate — COMMENTED** [review]
can you please revert the commit [`49602eb` (#6197)](https://github.com/juspay/hyperswitch/pull/6197/commits/49602ebc28e2009b363b6286f20302bd8f12250d)?

Lints formatting is fixed in the PR [6153](https://github.com/juspay/hyperswitch/pull/6153/files#diff-9873c474b804ded53ca57bd59d0424fbd661815501c77cf6449c9a60ba21cc96)

can you also please look after other comments as well? thanks!

**Comment by Ankesh2004** [comment]
@pixincreate  I reverted the eslint , and fixed all the minor issues mentioned in reviews , please look into it.

File: cypress-tests/cypress/support/commands.js
Comment (Ankesh2004) [reply cypress-tests/cypress/support/commands.js:2313]: @pixincreate Review now

File: cypress-tests/cypress/support/commands.js
Comment (pixincreate) [inline cypress-tests/cypress/support/commands.js:2331]: could you please add back these (blue color ones, not the red one) changes?

image

File: cypress-tests/cypress/support/commands.js
Comment (pixincreate) [reply cypress-tests/cypress/support/commands.js:2388]: add this back:

image

**Review by pixincreate — COMMENTED** [review]
we're almost done with the changes...

File: cypress-tests/cypress/support/commands.js
Comment (Ankesh2004) [reply cypress-tests/cypress/support/commands.js]: @pixincreate  All changes done

**Comment by Ankesh2004** [comment]
@pixincreate  Please review the changes

**Review by pixincreate — APPROVED** [review]
changes looks good.  
needs to be tested once.

**Comment by Ankesh2004** [comment]
> changes looks good. needs to be tested once.

So it is to be done from your side or do I have to test and submit results here?

**Comment by pixincreate** [comment]
yup, works!

image

**Comment by pixincreate** [comment]
lets wait for other's reviews and once done, the pr will be merged.

**Review by Sakilmostak — APPROVED** [review]

**Review by Gnanasundari24 — APPROVED** [review]

**Pull request merged**
